presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/TestPushProjectionThroughExchange.java

```diff
@@ -0,0 +1,156 @@
+/*
```

Comment:
Sure.

But note that test fails without the "changes" commit, as just-ported code does not ensure that Node after rewrite exposes same number of outputs.

Maybe we should squash all the three commits together. I left them separate mostly to make review easier.